private/bufpkg/bufconfig/buf_gen_yaml_file.go

```diff
@@ -481,8 +482,11 @@ type externalTypesConfigV1 struct {
 
 // externalBufGenYAMLFileV2 represents the v2 buf.gen.yaml file.
 type externalBufGenYAMLFileV2 struct {
-	Version string                           `json:"version,omitempty" yaml:"version,omitempty"`
-	Managed externalGenerateManagedConfigV2  `json:"managed,omitempty" yaml:"managed,omitempty"`
+	Version string                          `json:"version,omitempty" yaml:"version,omitempty"`
+	Managed externalGenerateManagedConfigV2 `json:"managed,omitempty" yaml:"managed,omitempty"`
+	// Clean, if set to true, will delete the output directories, zip files, or jar files
+	// before generation is run.
+	Clean   bool                             `json:"clean,omitempty" yaml:"clean,omitempty"`
```

Comment:
Since this is at the top level, I think we might need to make the key a little more specific. I know it looks a little longer, but just `clean` might not be very clear without referring to the docs. Maybe we should name it `clean_plugin_outs`? ^^"